traitsui/testing/tester/qt4/helpers.py

```diff
@@ -150,11 +156,7 @@ def mouse_click_qlayout(layout, index, delay):
     if not 0 <= index < layout.count():
         raise IndexError(index)
     widget = layout.itemAt(index).widget()
-    QTest.mouseClick(
-        widget,
-        QtCore.Qt.LeftButton,
-        delay=delay,
-    )
+    mouse_click_qwidget(widget, delay)
```

Comment:
one reason it would be a bit annoying to have instead made a separate `mouse_click_qbutton` function is that in places like here we may have a layout that contains buttons, or a layout containing other widgets.  In that cases here one would in theory need to have the isinstance check I added in `mouse_click_q_widget`.  
currently `mouse_click_qlayout` is only used for enum_editor and checklist_editor, and in both of those cases the layout is filled with buttons.  So right now we could just call a new `mouse_click_qbutton` function here, but I wasn't sure if there might be a use case for this function in which the layout wasn't populated with button.

Reply:
I'd suggest refactoring the first three lines that returns a widget (which occurs in a number of places), then  in the implementation of the CheckListEditor where it knows for sure the widget is a checkbox, call the refactored function to obtain the widget and then call the `click` method.
The code path we need for this handler needs to know it is dealing with checkboxes, the only reasonable place for that information to go it in the CheckListEditor.

Reply:
Ah no, scrap that. I did not see the `if isinstance(control, QtGui.QAbstractButton):` change above.